src/main/java/seedu/exercise/logic/commands/CustomRemoveCommand.java

```diff
@@ -30,51 +31,66 @@
         + "Example: " + COMMAND_WORD + " "
         + PREFIX_REMOVE_CUSTOM + "Rating ";
 
-    public static final String MESSAGE_SUCCESS = "Custom property removed: %1$s";
+    public static final String MESSAGE_SUCCESS_ALL_REMOVED = "Custom property removed: %1$s";
+    public static final String MESSAGE_SUCCESS_SINGLE_REMOVED = "%1$s removed for exercise %2$s";
     public static final String MESSAGE_FULL_NAME_NOT_FOUND = "This full name is not used by an "
         + "existing custom property";
 
+    private static final Logger logger = LogsCenter.getLogger(CustomRemoveCommand.class);
     private final String toRemove;
+    private final Optional<Index> index;
 
-    public CustomRemoveCommand(String toRemove) {
-        requireNonNull(toRemove);
+    public CustomRemoveCommand(String toRemove, Optional<Index> index) {
+        requireAllNonNull(toRemove, index);
         this.toRemove = toRemove;
+        this.index = index;
     }
 
     @Override
     public CommandResult execute(Model model) throws CommandException {
         requireNonNull(model);
 
-        if (!model.isFullNameUsedByCustomProperty(toRemove)) {
+        PropertyBook propertyBook = PropertyBook.getInstance();
+        if (!propertyBook.isFullNameUsedByCustomProperty(toRemove)) {
             throw new CommandException(MESSAGE_FULL_NAME_NOT_FOUND);
         }
 
-        model.removeCustomProperty(toRemove);
-        updateCustomPropertiesOfAllExercises(model);
-        return new CommandResult(String.format(MESSAGE_SUCCESS, toRemove));
+        if (index.isEmpty()) {
+            logger.info("Removing " + toRemove + " from the app");
+            propertyBook.removeCustomProperty(toRemove);
+            updateCustomPropertiesOfAllExercises(model);
+            return new CommandResult(String.format(MESSAGE_SUCCESS_ALL_REMOVED, toRemove));
+        } else {
+            logger.info("Removing " + toRemove + " from a single exercise");
+            Index indexToRemove = index.get();
+            updateCustomPropertiesOfSingleExercise(model, indexToRemove);
+            return new CommandResult(String.format(MESSAGE_SUCCESS_SINGLE_REMOVED, toRemove,
+                indexToRemove.getOneBased()));
```

Comment:
Great naming and split here. really readable 👍

Reply:
Thank you boss 👍